crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -28,8 +28,9 @@ pub struct DynamicScene {
 
 /// A reflection-powered serializable representation of an entity and its components.
 pub struct DynamicEntity {
-    /// The transiently unique identifier of a corresponding `Entity`.
-    pub entity: u32,
+    /// The identifier of the entity, unique within a scene (and the world it may have been generated from).
+    /// Components containing entitiy references will identify their referant via this identifier.
+    pub entity: Entity,
```

Comment:
Nit: I know there's talks of "scene entity" vs "real entity". Does this entity fall into either of those? Does it not matter and/or is it not guaranteed?

If it does matter, might be worth adding a doc comment on it.

Reply:
`Entity` is just an identifier, and the only constraint here is that it must be consistent with the references that exist within components in the scene. I guess that makes it a "scene entity" though scenes themselves are sort of like worlds that organize their entity data differently than a `World`. That constraint is what I was trying to indicate with the sentence you commented on above.